packages/leap/lib/src/leap_game.dart

```diff
@@ -46,10 +66,23 @@ class LeapGame extends FlameGame with HasTrackedComponents {
     AssetBundle? bundle,
     Images? images,
     Map<String, TiledObjectHandler> tiledObjectHandlers = const {},
+    LeapMapTransition Function(LeapGame) mapTransitionFactory =
+        LeapMapTransition.defaultFactory,
   }) async {
+    final currentMap = _leapMap;
+    LeapMapTransition? transitionComponent;
+    if (currentMap != null) {
+      onMapUnload();
+
+      final transition = transitionComponent = mapTransitionFactory(this);
+      camera.viewport.add(transition);
+      await transition.newMapReadyToBeLoaded;
```

Comment:
naming nit: `newMapReadyToBeLoaded` makes me think we're waiting on the map, but really I think this is waiting on the transition intro to finish, so maybe this future should be called `transition.introFinished` or something?

Then the other future could be `outroFinished`

Reply:
oh, I like those names, also, I've never heard about the `outro` world before, at first I thought that was a typo or something but it is really a word hahaha!

Fun fact, in portuguese outro means another hahaha

Reply:
Haha, it's an odd word, I'm 99% sure someone made it up a long time ago just because "out" is the opposite of "in" and glued the words together

`outro` makes sense as another, it's like `otro` in spanish :)